scripts/pr_review_test.py

```
@@ -1265,6 +1267,569 @@ def test_every_secret_the_guide_requires_is_one_the_workflow_accepts(self) -> No
+        self.assertEqual(verified, [], "the first decision for an index wins")
+
+    def test_an_unhashable_verdict_does_not_crash_the_run(self) -> None:
+        # A set-membership test on model output raises TypeError for any JSON
+        # array or object. TypeError is not ModelOutputError, so it escaped the
+        # handler written for bad model output entirely, and the publish-on-exit
+        # path then reported a verdict derived from state that never recorded
+        # the failure.
+        for verdict in (["keep"], {"v": "keep"}, 7, None):
+            with self.subTest(verdict=verdict):
+                payload = {"findings": [{"index": 0, "verdict": verdict, "reason": "r"}]}
+                with self.assertRaises(pr_review.ModelOutputError):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)